### PR TITLE
feat(api-docs): Add external issue endpoints

### PR DESCRIPTION
**Pull request by ceorourke**
Add the endpoint documentation for the following endpoints in a new sidebar item:

GET `organizations/org.slug/sentry-app-installations/`
POST `sentry-app-installations/install.uuid/external-issues/`
DELETE `sentry-app-installations/install.uuid/external-issues/external_issue.id/`

Screen Shot 2021-02-11 at 12 53 36 PM
Screen Shot 2021-02-11 at 12 56 16 PM
Screen Shot 2021-02-11 at 12 56 25 PM
Screen Shot 2021-02-11 at 12 56 06 PM

### CODE REVIEW

**Comment by ceorourke** [comment]
I put these under the organizations section in the API docs but I could make a new one for Internal Integrations if that makes more sense.

**Comment by manuzope** [comment]
@ceorourke how about a new section called `Integration Platform`?

**Comment by manuzope** [comment]
Also, I see you have a description for the body params in "Create an external issue". Is that not showing up in the docs?

**Comment by ceorourke** [comment]
@manuzope yeah I don't think any body param descriptions are showing up

**Comment by manuzope** [comment]
:+1: Created a follow up ticket

**Comment by MeredithAnya** [comment]
I think the GET endpoint for the sentry-app-installations will show all the installations, regardless of if they are internal or not

**Comment by ceorourke** [comment]
@MeredithAnya ooh good point, in my head I think of all sentry apps as "internal integrations" even though that's not true. If we add the status do you see that being in a separate PR and then we can update the docs?

**Comment by ceorourke** [comment]
@MeredithAnya that's a good catch - it looks a little inconsistent too like [here](https://docs.sentry.io/api/events/update-an-issue/) the path param is `issue_id` but the description says group.

File: src/sentry/api/serializers/models/platformexternalissue.py [hunk 1]
```diff
@@ -7,7 +7,7 @@ class PlatformExternalIssueSerializer(Serializer):
     def serialize(self, obj, attrs, user):
         return {
             "id": str(obj.id),
-            "groupId": str(obj.group_id),
+            "issueId": str(obj.group_id),
```
Comment (MeredithAnya) [inline src/sentry/api/serializers/models/platformexternalissue.py:10]: we should update https://github.com/getsentry/sentry/blob/264bc5f4d369cd9a6bed2ff18ec52b50789eeba6/src/sentry/static/sentry/app/types/index.tsx#L1260
and make sure that any frontend changes are made that need to be. I actually don't think we use the group id from this, (I think we have the actual group as a prop) but might wanna double check

**Review by MeredithAnya — CHANGES_REQUESTED** [review]

File: src/sentry/api/serializers/models/platformexternalissue.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/api/serializers/models/platformexternalissue.py:10]: @MeredithAnya cool I updated that and poked around, the only other thing I saw was https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/actionCreators/platformExternalIssues.tsx#L9 but that's for a different endpoint anyway.

**Review by MeredithAnya — APPROVED** [review]
LGTM (just a couple tests to fix up)

**Pull request merged**